superduperdb/ext/pillow/encoder.py

```diff
@@ -87,3 +87,25 @@ def image_type(
     identifier='pil_image_hybrid',
     encodable='artifact',
 )
+
+
+class PilDataTypeFactory(DataTypeFactory):
+    """A factory for pil image # noqa."""
+
+    @staticmethod
+    def check(data: t.Any) -> bool:
```

Comment:
Would it make sense(future) to add a function `check=my_checking_function` to `DataType`?